### PR TITLE
Feature/541/Added types to ViewSwitcher component

### PR DESCRIPTION
**Pull request by mxrcury**
Typing ViewSwitcher component with TypeScript

### CODE REVIEW

File: src/components/view-switcher/ViewSwitcher.tsx [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
 
 import FormatListBulletedIcon from '@mui/icons-material/FormatListBulleted'
 import GridViewIcon from '@mui/icons-material/GridView'
 
-import { cardsViews } from '~/constants'
 import { styles } from '~/components/view-switcher/ViewSwitcher.styles'
 
-const ViewSwitcher = ({ setOffersView, offersView }) => {
-  const changeOffersView = (_event, view) => setOffersView(view)
+import { CardsViewEnums } from '~/types/findOffers/enums/viewSwitcher.enums'
```
Comment (Tolik170) [inline src/components/view-switcher/ViewSwitcher.tsx:11]: Now you had better to import all type declarations from '~types', look at structure and other files

File: src/components/view-switcher/ViewSwitcher.tsx [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+import { CardsViewTypes } from '~/types/findOffers/types/viewSwitcher.types'
+
+interface ViewSwitcherProps {
+  setOffersView:(value: CardsViewTypes) => void
```
Comment (Tolik170) [inline src/components/view-switcher/ViewSwitcher.tsx:15]: add space after :

Comment (mxrcury) [reply src/components/view-switcher/ViewSwitcher.tsx]: added

File: src/components/view-switcher/ViewSwitcher.tsx (hunk 1 above)
Comment (mxrcury) [reply src/components/view-switcher/ViewSwitcher.tsx]: changed imports of types

File: src/components/view-switcher/ViewSwitcher.tsx [hunk 3]
```diff
@@ -1,32 +1,40 @@
...
-const ViewSwitcher = ({ setOffersView, offersView }) => {
-  const changeOffersView = (_event, view) => setOffersView(view)
+import { CardsViewEnums, CardsViewTypes } from '~/types'
+
+interface ViewSwitcherProps {
+  setOffersView: (value: CardsViewTypes) => void
+  offersView: CardsViewTypes
+}
+
+const ViewSwitcher: FC<ViewSwitcherProps> = ({ setOffersView, offersView }) => {
```
Comment (MickhaelSh) [inline src/components/view-switcher/ViewSwitcher.tsx:18]: name props as value and onChange

Comment (mxrcury) [reply src/components/view-switcher/ViewSwitcher.tsx]: ok, changed

**Review by maarekua — APPROVED** [review]

**Pull request merged**